lib/folio/holdings.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Folio
+  class Holdings
+    YEAR_REGEX = /(?:19|20)\d{2}/
+    # We've seen chronologies that looks like full dates, MON YYYY, WIN YYYY, and nil
+    def self.find_latest(holdings)
+      # NOTE: We saw some piece records without 'chronology'. Was this just test data?
+      pieces = holdings.filter_map do |piece|
+        piece.merge(sortable_date: sortable_date(piece),
+                    sortable_tokens: sortable_tokens(piece))
+      end
+      pieces.max_by { |piece| [piece.fetch(:sortable_date)] + piece.fetch(:sortable_tokens) }
+    end
+
+    # @return [Date] a date derived from chronology, enumeration or a really old date (sorts to back) if none are found
+    def self.sortable_date(piece)
+      extract_sortable_date(piece['chronology']) ||
+        extract_sortable_date(piece['enumeration']) ||
+        Date.parse('1000-01-01')
+    end
+
+    # @return [Array<String>] a list of tokens derived from chronology and enumeration. Numbers are favored over strings.
+    def self.sortable_tokens(piece)
+      Array(
+        extract_sortable_tokens(piece['chronology']) ||
+          extract_sortable_tokens(piece['enumeration']) ||
+          piece['enumeration']
```

Comment:
Isn't there still a problem if we try to sort between the "sortable tokens" and strings?

```
[Array('1'), Array(1)].sort
(irb):4:in `sort': comparison of Array with Array failed (ArgumentError)
```

Reply:
Okay, I've added that test and made it pass.